### ISSUE TITLE
`@stylistic/styled/unicode-bom` reads a mark opening a template as the byte-order mark of the file

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#728]
`@stylistic/styled/unicode-bom` reads the mark PostCSS takes off the head of each template as the byte-order mark of the file. Under `postcss-styled-syntax` every template is a root of its own, parsed through a `postcss.Input` of its own, and that input's `hasBOM` says whether the template's content opens with U+FEFF. The rule passes over a root carrying `document`, which is how it leaves a `<style>` block of `postcss-html` alone, but a template's root hangs in its document as `parent` and carries no `document`, so the rule reads it.

So the rule answers once per template, at the template, about a character of JavaScript rather than about the file:

```
always  "﻿let x = css`a {}`;"                        → 1:14 Expected Unicode BOM
always  "﻿let x = css`a {}`;\nlet y = css`b {}`;"    → 1:14 and 2:13 Expected Unicode BOM
always  "let x = css`﻿a {}`;"                        → nothing
never   "let x = css`﻿a {}`;"                        → 1:13 Unexpected Unicode BOM
never   "﻿let x = css`a {}`;"                        → nothing
```

A file opening with the mark is told under `always` that it lacks one, once per template, and a file without it passes as soon as a template opens with U+FEFF; under `never` a template opening with U+FEFF is reported, while the file's own mark is not. Under `postcss-html` both spellings, the mark in front of `<style>` and inside it, draw nothing under `never`, as the rule intends.

Measured on `7af36fb` with `postcss-styled-syntax` 0.7.2. Fixing the syntax's loss of that mark (hudochenkov/postcss-styled-syntax#40) changes none of these rows.

## What would end it

Passing over a root whose parent is a document too, as the rule already does for one carrying `document`. This also comes before #700: a fixer setting `hasBOM` on a template's root would write U+FEFF into the template once the syntax prints the flag back.
